Dp 6 integrate employee endpoints

moe1011 opened the pull request:
Updated functionality of multiple Employee Classes and Methods

- **EmployeeController**
-- getEmployees now authenticates the user and accepts a storeId then sends it to the EmployeeService for further checking
-- getEmployee now authenticates the user and accepts a storeId then sends it to the EmployeeService for further checking
-- newEmployee now authenticates the user and accepts a storeId and password then sends it to the EmployeeService for further checking

- **EmployeeService**
-- getEmployees uses the EmployeeRepository to find all employees by the given storeId, if the store contains employees then they will be returned, otherwise a ResponseStatusException of Not Found will be thrown.
-- getEmployee uses the EmployeeRepository to find a specific employee given the storeId and employeeId, if the store finds the employee then they will be returned, otherwise a ResponseStatusException of Not Found will be thrown.
-- newEmployee checks if the store exists by using the StoreRepository and storeId, if it does it will encrypt the employees password, set the newly encrypted password to employee then save the newly created employee to the database along with returning the employee. If a store is not found a Not Found exception will be throw. If an illegal password is used a Bad Request exception will be thrown. _(Bakr Matlab and I both contributed in setting up the newEmployee method via discord messaging and calls)_

- **EmployeeRepository**
-- findEmployeesByStore_Id, Finds an employee from the database given the storeId and return a List of Employees
-- findEmployeeByStore_IdAndId, Finds an employee from the database given the storeId and employeeId and returns a singular Employee

roynulrohan committed:
Created Employee Model. Setup boilerplate TODO files and methods

moe1011 committed:
For EmployeeService and EmployeeRepositroy
- Updated functionality for getEmployees method
- Added a new repository to fine employees by store ID

moe1011 committed:
For EmployeeService and EmployeeRepositroy
- Updated functionality for getEmployees method
- Added a new repository to fine employees by store ID

bakrmatlab committed:
Created endpoints for the following methods
-getEmployees
-getEmployee
-newEmployee

moe1011 committed:
Updated Employee Controller, Repository and Service
- getEmployee and newEmployee Methods
- Added a repository to fetch an Employee by store ID and employee ID

moe1011 committed:
Fixed an issue with the EmployeeService and EmployeeController
- Employee wasn't being assigned a store
- Methods weren't properly returning storeId

moe1011 committed:
Added partial comments and javadoc for the Employee classes

roynulrohan commented:
Good work. Not merging PR until the following conditions are met:

- Remove all javadocs and unnecessary comments such as the date heading in `EmployeeService.java`, normal comments can stay.
- Within `EmployeeController.java`, an Authentication object is being declared in all mappings but are unused. Please remove them.

bakrmatlab committed:
Fixed issue in EmployeeController meathod
-Removed Authentication object from EmployeeController

bakrmatlab committed:
Removed Authentication object and javadocs.
-Removed Authentication object from EmployeeController.java
-Removed all javadocs and unnecessary comments

roynulrohan committed:
Fixed up some formatting
